src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +137,1039 @@ static DamageTable D_80A63CE0 = {
+            this->timer += 10;
+        }
+        this->skyFloatPhase += 0x320;
+    } else {
+        EnBal_SetupFloatIdle(this);
+    }
+}
+
+void EnBal_SetupGroundIdle(EnBal* this) {
+    if (this->actionFunc == EnBal_Fall) {
+        this->forceEyesShut = false;
+        this->eyeTexIndex = TINGLE_EYETEX_OPEN;
+        this->idleAnimStage = TINGLE_IDLESTAGE_WAIT;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_IDLE);
+    } else if (Rand_Next() & 1) {
+        this->idleAnimStage = TINGLE_IDLESTAGE_ACTIVITY;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK);
+    } else {
+        this->idleAnimStage = TINGLE_IDLESTAGE_ACTIVITY;
+        this->forceEyesShut = true;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_SPIN);
+    }
+
+    this->watchTarget = TINGLE_WATCH_TARGET_NONE;
+    if (this->locationMapId == TINGLE_MAP_CLOCK_TOWN) {
+        if (!gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+            // Effectively turn off reinflation timer by setting above 300
+            this->timer = 301;
+        } else if (this->inflateEarly == true) {
+            // If this field is set, reduce time to balloon reinflation.
+            this->inflateEarly = false;
+            this->timer = 290;
+        } else {
+            this->timer = 0;
+        }
+    } else {
+        this->timer = 0;
+    }
+
+    this->actionFunc = EnBal_GroundIdle;
+}
+
+void EnBal_GroundIdle(EnBal* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (this->timer == 300) {
+        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            EnBal_SetupInflateBalloon(this);
+            return;
+        }
+    } else if (this->timer < 300) {
+        this->timer++;
+    }
+
+    if (Actor_ProcessTalkRequest(&this->picto.actor, &play->state)) {
+        this->forceEyesShut = false;
+        this->eyeTexIndex = TINGLE_EYETEX_OPEN;
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TALKED_TINGLE)) {
+            if (GET_PLAYER_FORM == EnBal_GetRecognizedPlayerForm()) {
+                this->watchTarget = TINGLE_WATCH_TARGET_PLAYER;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK);
+                Message_StartTextbox(play, 0x1D0C, &this->picto.actor);
+                this->textId = 0x1D0C;
+            } else {
+                this->watchTarget = TINGLE_WATCH_TARGET_PLAYER;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK);
+                Message_StartTextbox(play, 0x1D05, &this->picto.actor);
+                this->textId = 0x1D05;
+            }
+
+            if (!gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+                // Reinflation should be unreachable while player does not have magic,
+                //  so I am not sure why this is here?
+                this->inflateEarly = true;
```

Comment:
```suggestion
                // Reinflation should be unreachable while player does not have magic
                this->inflateEarly = true;
```
I think this comment is enough